storage/innobase/lock/lock0lock.cc

```diff
@@ -584,141 +597,177 @@ ATTRIBUTE_NOINLINE static bool wsrep_is_BF_lock_timeout(const trx_t &trx)
+       the waiting next-key lock.
 
-		Also, insert intention locks do not disturb each
-		other. */
+       Also, insert intention locks do not disturb each
+       other. */
 
-		return false;
-	}
+    return false;
+  }
 
 #ifdef HAVE_REPLICATION
-	if ((type_mode & LOCK_GAP || lock2->is_gap())
-	    && !thd_need_ordering_with(trx->mysql_thd, lock2->trx->mysql_thd)) {
-		/* If the upper server layer has already decided on the
-		commit order between the transaction requesting the
-		lock and the transaction owning the lock, we do not
-		need to wait for gap locks. Such ordeering by the upper
-		server layer happens in parallel replication, where the
-		commit order is fixed to match the original order on the
-		master.
-
-		Such gap locks are mainly needed to get serialisability
-		between transactions so that they will be binlogged in
-		the correct order so that statement-based replication
-		will give the correct results. Since the right order
-		was already determined on the master, we do not need
-		to enforce it again here.
-
-		Skipping the locks is not essential for correctness,
-		since in case of deadlock we will just kill the later
-		transaction and retry it. But it can save some
-		unnecessary rollbacks and retries. */
+  if ((type_mode & LOCK_GAP || lock2->is_gap())
+      && !thd_need_ordering_with(trx->mysql_thd, lock2->trx->mysql_thd))
+  {
+    /* If the upper server layer has already decided on the
+       commit order between the transaction requesting the
+       lock and the transaction owning the lock, we do not
+       need to wait for gap locks. Such ordeering by the upper
+       server layer happens in parallel replication, where the
+       commit order is fixed to match the original order on the
+       master.
+
+       Such gap locks are mainly needed to get serialisability
+       between transactions so that they will be binlogged in
+       the correct order so that statement-based replication
+       will give the correct results. Since the right order
+       was already determined on the master, we do not need
+       to enforce it again here.
+
+       Skipping the locks is not essential for correctness,
+       since in case of deadlock we will just kill the later
+       transaction and retry it. But it can save some
+       unnecessary rollbacks and retries. */
 
-		return false;
-	}
+    return false;
+  }
 #endif /* HAVE_REPLICATION */
 
 #ifdef WITH_WSREP
-	/* New lock request from a transaction is using unique key
-	scan and this transaction is a wsrep high priority transaction
-	(brute force). If conflicting transaction is also wsrep high
-	priority transaction we should avoid lock conflict because
-	ordering of these transactions is already decided and
-	conflicting transaction will be later replayed. */
-	if (trx->is_wsrep_UK_scan()
-	    && wsrep_thd_is_BF(lock2->trx->mysql_thd, false)) {
-		return false;
-	}
+  if (trx->is_wsrep())
+  {
```

Comment:
It would be better to avoid reformatting this entire function when the essential change is here. I would suggest to add a separate `static ATTRIBUTE_NOINLINE ATTRIBUTE_COLD` function for the Galera specific code, and invoke it at the end of the function:
```c++
#ifdef WITH_WSREP
	if (trx->is_wsrep()) {
		return lock_rec_has_to_wait_wsrep(…);
	}
#endif
	return true;
```

Reply:
Okay